File: typescript/src/bebop_handlers/gameOver.ts
import GameState from '../classes/gameState';
import { GameOverMsg } from '../models/pactheman.models';
import gameOverMsgToHumanReadable from '../utils/extensions/gameOver';
import BebopHandler from './bebopHandler';

export default class GameOverMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = GameOverMsg.decode(message)
        console.log(
            `Received GameOver because ${gameOverMsgToHumanReadable(message)}. Ending lean client now.`);
        GameState.Instance.setGameOver();
        GameState.Instance.signalGameOver(message);
    }
}

File: dart/lib/bebop_handlers/game_over.dart
import 'package:lean_dart_client/classes/game_state.dart';

import 'bebop_handler.dart';
import 'package:lean_dart_client/schemas.g.dart';

class GameOverMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    message = GameOverMsg.decode(message);
    print(
        'Received GameOver because ${message.toHumanReadable()}. Ending lean client now.');
    GameState.instance.signalGameOver(message);
  }
}
